src/imshow.jl

```diff
@@ -46,11 +50,8 @@ function imshow(
     # otherwise, use our own implementation
     io_h, io_w = maxsize
     img_h, img_w = map(length, axes(img))
-    str = if img_h <= io_h-4 && 2img_w <= io_w
-        first(encodeimg(BigBlocks(),   colordepth, img, io_h-4, io_w))
-    else
-        first(encodeimg(SmallBlocks(), colordepth, img, io_h-4, io_w))
-    end
+    enc = img_h <= io_h - 4 && 2img_w <= io_w ? BigBlocks : SmallBlocks
+    str = first(encodeimg(enc(), colordepth, img, io_h - 4, io_w))
```

Comment:
The `imshow` implementation is deeply involved with the ASCII encoder so it should also be moved to the backends. Except for the `use_sixel(img)` part, this function is quite like `sixel_encode([io], img; kwargs...)`.

The `use_sixel(img) && return sixel_encode(io, img)` part in all the three `imshow` methods should be still kept on the frontend side in the mean time.

Reply:
I don't understand this, in your picture, it is still part of the frontend.

Reply:
Sorry, my figure didn't get very concise. Let me try to rephrase the words. The entire pipeline of the current `imshow` consists of three stages:

1. `imshow` decides the best block size: is it BigBlocks or SmallBlocks? 
2. `encodeimg` decides the bit depth: is it 8bit or 24bit?
3. `colorant2ansi` encodes the pixel

As the only interface that users are supposed to use we can let `ascii_encode` contain all three stages. We'd want to do this because these stages are not related to `sixel_encoding` at all.

Reply:
Thanks for the clarification. I gave it another shot, tell me what you think.
I haven't carefully checked yet if the changes in the reference files are valid.